### DISCUSSION TITLE
Create the simplest RPG-like game that showcases full potential of RAUI in games

### DISCUSSION THREAD

**Discussion by PsichiX** [RAUI-labs/raui#72] [Ideas]
While book is a good step towards making users getting more familiar with RAUI, we kinda lack of the real examples of how to use RAUI in games - we have added a lot of improvements to ease making UI in games, but now i think we need some small mini game that will showcase how to design production-ready UI. For that might turn `in-game` demo into simple mini game or even make a separate demo for that mini game.

I'll consider this thread a discussion about ideas of what and how we are gonna design that simple RPG game. We are gonna discuss each problem in separate posts and we will reply to them.

**Comment by PsichiX** [comment]
# Pick rust game engine that will be used for the game.

right now we support few game engines:
- ggez - still waits for new maintainer so its integration is old and does not cover all RAUI features;
- tetra - it is used already for demos, but similar to ggez it doesn't support all RAUI features yet;
- bevy-retro - i believe @zicklag has the most features covered, also users are getting more and more familiar with bevy, but i believe its license is something that keeps them out of bevy-retro (not sure if that's something that will change @zicklag ?)
- oxygengine - it's web only and it doesn't get updates of RAUI as fast as other supported engines

There are few engines that we could try to make official integrations, for example macroquad or bevy?

**Reply by zicklag** [reply]
> bevy-retro - i believe @zicklag has the most features covered, also users are getting more and more familiar with bevy, but i believe its license is something that keeps them out of bevy-retro (not sure if that's something that will change @zicklag ?)


The license isn't someting that will be changing. You can see https://github.com/katharostech/bevy_retro/issues/1#issuecomment-816992576 for the explanation.

**Reply by PsichiX** [reply]
ok, just wanted to be sure, have no problem with the decision.
